src/components/TokenBalance.tsx

```diff
@@ -301,31 +283,22 @@ export function HomeTokenBalance() {
 export function FiatExchangeTokenBalance() {
   const { t } = useTranslation()
   const totalBalance = useTotalTokenBalance()
-  const tokenBalances = useTokensWithTokenBalance()
 
   const onViewBalances = () => {
     ValoraAnalytics.track(FiatExchangeEvents.cico_landing_token_balance, {
       totalBalance: totalBalance?.toString(),
     })
-    navigate(
-      getFeatureGate(StatsigFeatureGates.SHOW_ASSET_DETAILS_SCREEN)
-        ? Screens.Assets
-        : Screens.TokenBalances
-    )
+    navigate(Screens.Assets)
   }
 
   return (
     <View style={styles.container} testID="FiatExchangeTokenBalance">
       <View style={styles.titleExchange}>
         <View style={styles.row}>
-          {tokenBalances.length > 1 ? (
```

Comment:
This behavior was updated because the tests were coupled together. Now the behavior of the home token balances and the fiat exchange token balances matches.

Reply:
The behavior is different when show_asset_details_screen was set to true which is 100%. If we want to keep the behavior of the existing FiatExchangeTokenBalance some tests will need to be updated. Due to the coupled testing it was unclear if we ever expected these two components to diverge in behavior like they have.

Reply:
I believe the FiatExchangeTokenBalance behavior has always been different from the HomeTokenBalance behavior. 
FiatExchangeTokenBalance shows the ViewBalances touchable if num tokens > 1 both before and after the asset details change. HomeTokenBalance before asset details only included the ViewBalances touchable if num tokens >= 1 and after asset details, it is always included (since we also show tokens with zero balance in the assets screen).

I think we had coupled tests prior just because the case for 0 tokens overlapped. Can't we just drop the [`it.each`](https://github.com/valora-inc/wallet/blob/f0c74d56180c6f336f9884c2fec3480e92af96fb/src/components/TokenBalance.test.tsx#L199) and just keep this test for `FiatExchangeTokenBalance` and keep [this test](https://github.com/valora-inc/wallet/blob/f0c74d56180c6f336f9884c2fec3480e92af96fb/src/components/TokenBalance.test.tsx#L223) for HomeTokenBalance and just remove the feature gate in the name and mocks?

Reply:
Fixed in 62c8ea5201f4d31e467122df8b60f22f5ceedbcc and tests updated in 99c38dcbe8a4679859fef4da97d01ec008ab911c!